deploy/manifests/virtual-kubelet/base/statefulset.yaml

```diff
@@ -1,17 +1,30 @@
-apiVersion: apps/v1
-kind: Deployment
+apiVersion: v1
```

Comment:
What's the purpose of the virtual-kubelet Service?

Reply:
So when using a statefulset `serviceName` is a required field for it, so we need a headless service to be included as well.

According to Kubernetes documentation "StatefulSets currently require a Headless Service to be responsible for the network identity of the Pods. You are responsible for creating this Service."

The above can be found in this document in the "Limitations" section.
https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/